@xen-orchestra/lite/src/components/Tree.tsx

```diff
@@ -96,43 +106,79 @@ const CustomContent = React.forwardRef(function CustomContent(props: CustomConte
   )
 })
 
-const renderItem = ({ children, id, label, to, tooltip }: ItemType) => {
+const renderItem = ({ children, id, label, to, tooltip }: ItemType, defaultSelectedNodes?: Array<string>) => {
   return (
     <TreeItem
       ContentComponent={CustomContent}
       // FIXME: ContentProps should only be React.HTMLAttributes<HTMLElement> or undefined, it doesn't support other type.
       // when https://github.com/mui-org/material-ui/issues/28668 is fixed, remove 'as CustomContentProps'.
-      ContentProps={{ to } as CustomContentProps}
+      ContentProps={{ defaultSelectedNodes, to } as CustomContentProps}
       label={tooltip ? <Tooltip title={tooltip}>{label}</Tooltip> : label}
       key={id}
       nodeId={id}
     >
-      {Array.isArray(children) ? children.map(renderItem) : null}
+      {Array.isArray(children) ? children.map(item => renderItem(item, defaultSelectedNodes)) : null}
     </TreeItem>
   )
 }
 
 const Tree = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
   {
-    initialState: ({ defaultSelectedNodes }) => ({
-      selectedNodes: defaultSelectedNodes === undefined ? [] : defaultSelectedNodes,
-    }),
+    initialState: ({ collection, defaultSelectedNodes }) => {
+      if (defaultSelectedNodes === undefined) {
+        return {
+          expandedNodes: [collection[0].id],
+          selectedNodes: [],
+        }
+      }
+
+      // expandedNodes should contain all nodes up to the defaultSelectedNodes.
+      let expandedNodes: Array<string> = []
+      const tempExpandedNodes: Set<string> = new Set()
+      const addExpandedNode = (collection: Array<ItemType> | undefined) => {
+        if (collection === undefined) {
+          return
+        }
+
+        for (const node of collection) {
+          const children = node.children
+          if (children === undefined) {
+            if (defaultSelectedNodes.includes(node.id)) {
+              expandedNodes = expandedNodes.concat(Array.from(tempExpandedNodes))
+            }
+          } else {
+            tempExpandedNodes.add(node.id)
+            addExpandedNode(node.children)
+            tempExpandedNodes.delete(node.id)
```

Comment:
Wouldn't this add the parent node multiple times if multiple children are selected?

Reply:
No, it adds the parent node only once. `addExpandedNode(node.children)` checks/adds children.